### PR TITLE
feat / add connection status to gateway flag

### PR DESCRIPTION
**Pull request by vic-en**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
[ch25670]

**Tests performed by the developer**:
Started client with gateway off, flag was `STOPPED, OFFLINE`
Started gateway, flag was `RUNNING, OFFLINE`
after a while, flag was `RUNNING, ONLINE`

**Tips for QA testing**:
- Start client with gateway off, flag should be `STOPPED, OFFLINE`
- Started gateway, flag should be `RUNNING, OFFLINE`
- after a while, flag should be `RUNNING, ONLINE`

### CODE REVIEW

File: hummingbot/core/gateway/status_monitor.py [hunk 1]
```diff
@@ -69,9 +75,14 @@ async def _monitor_loop(self):
                         GATEWAY_CONNECTORS.extend([connector["name"] for connector in gateway_connectors.get("connectors", [])])
 
                         await self.update_gateway_config_key_list()
+                    elif self._current_connector_conn_status is Status.OFFLINE:
+                        gateway_connectors_status = await GatewayHttpClient.get_instance().get_gateway_status(fail_silently=True)
+                        self._current_connector_conn_status = Status.ONLINE if len(gateway_connectors_status) > 0 and \
+                            gateway_connectors_status[0]["currentBlockNumber"] > 0 else Status.OFFLINE
```
Comment (CrimsonJacket) [inline hummingbot/core/gateway/status_monitor.py:81]:
```suggestion
                        self._current_connector_conn_status = Status.ONLINE \
                            if any([status["currentBlockNumber"] > 0 for status in gateway_connectors_status]) else Status.OFFLINE
```

**Review by CrimsonJacket — CHANGES_REQUESTED** [review]
Minor recommendation.

**Comment by PtrckM** [comment]
- update configuration using `gateway config {key}`  gateway status will stay **RUNNING, OFFLINE**
  - restarting the client will get the same result even wallet already added.

https://user-images.githubusercontent.com/54516858/164152611-b567c6dc-9039-4bf0-b897-90699a0711ff.mov

**Comment by vic-en** [comment]
> * update configuration using `gateway config {key}`  gateway status will stay **RUNNING, OFFLINE**
>   
>   * restarting the client will get the same result even wallet already added.
> 
>  status_offline.mov

Right, that's expected. So,
STOPPED, OFFLINE implies gateway isn't running and def not connected to any node.
RUNNING, OFFLINE implies gateway has been stated but hasn't connected to any node. Hence, it is offline.
RUNNING, ONLINE implies gateway has started and is communicating to one or more node.

Actions that make a gateway instance online are:
- Running balance command
- Starting any strategy that communicates with gateway.
Basically any call to endpoint or action and make gateway to interact with a node.

**Review by RC-13 — APPROVED** [review]
LGTM! Thank you Victor for confirming the behavior.

Test summary:
- Clone and install the feature branch and set up the Hummingbot client
- Confirm new gateway status and confirmed that the status is STOPPED, OFFLINE when no gateway config files or if the gateway container is stopped.
- Create a gateway docker container and confirmed that the gateway status changed to RUNNING, OFFLINE.
- Connect the gateway connector wallet and start an AMM_arb strategy, this changes the gateway status to RUNNING, ONLINE
- Configure gateway parameter to restart the container, this changes the gateway status to RUNNING, OFFLINE
- Running balance or restarting the strategy will trigger the communication with the node and changing the status to RUNNING, ONLINE
